gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -309,91 +329,119 @@ private static void ignoreFromCheckUnusedDependencies(Project proj) {
-                    task.setDescription("Runs `npm tsc` to compile generated TypeScript files into JavaScript files.");
-                    task.setGroup(TASK_GROUP);
-                    task.commandLine(npmCommand, "run-script", "build");
-                    task.workingDir(srcDirectory);
-                    task.dependsOn(installTypeScriptDependencies);
-                    task.getOutputs().dir(srcDirectory);
-                });
-                Task publishTypeScript = project.getTasks().create("publishTypeScript", Exec.class, task -> {
-                    task.setDescription("Runs `npm publish` to publish a TypeScript package "
-                            + "generated from your Conjure definitions.");
-                    task.setGroup(TASK_GROUP);
-                    task.commandLine(npmCommand, "publish");
-                    task.workingDir(srcDirectory);
-                    task.dependsOn(compileConjureTypeScript);
-                    task.dependsOn(compileTypeScript);
-                });
-                subproj.afterEvaluate(
-                        _p -> subproj.getTasks().maybeCreate("publish").dependsOn(publishTypeScript));
-                Task cleanTask = project.getTasks().findByName(TASK_CLEAN);
-                cleanTask.dependsOn(project.getTasks().findByName("cleanCompileConjureTypeScript"));
+                TaskProvider<Exec> compileTypeScript = project.getTasks()
+                        .register("compileTypeScript", Exec.class, task -> {
+                            task.setDescription(
+                                    "Runs `npm tsc` to compile generated TypeScript files into JavaScript files.");
+                            task.setGroup(TASK_GROUP);
+                            task.commandLine(npmCommand, "run-script", "build");
+                            task.workingDir(srcDirectory);
+                            task.dependsOn(installTypeScriptDependencies);
+                            task.getOutputs().dir(srcDirectory);
+                        });
+                TaskProvider<Exec> publishTypeScript = project.getTasks()
+                        .register("publishTypeScript", Exec.class, task -> {
+                            task.setDescription("Runs `npm publish` to publish a TypeScript package "
+                                    + "generated from your Conjure definitions.");
+                            task.setGroup(TASK_GROUP);
+                            task.commandLine(npmCommand, "publish");
+                            task.workingDir(srcDirectory);
+                            task.dependsOn(compileConjureTypeScript);
+                            task.dependsOn(compileTypeScript);
+                        });
+                linkPublish(subproj, publishTypeScript);
             });
         }
     }
 
+    private static void linkPublish(Project project, TaskProvider<?> depTask) {
```

Comment:
I'm not sure I fully understand the benefit of this method over `maybeCreate`. In the end we still create a publish task if it doesn't exist yet so the number of created tasks shouldn't change?

Reply:
one is a TaskProvider though, so config happens later. This doesn't matter for that individual task all that much, but it is a cascade effect. If that task is instantiated, 7 other tasks are also eagerly configured in each project, including the conjure compile tasks. It adds up.